.github/workflows/qemu.yaml

```diff
@@ -0,0 +1,46 @@
+name: qemu
+
+on:
+  push:
+    branches:
+      - master
+      - 'release/**'
+  pull_request:
+
+permissions:
+  contents: read
+
+jobs:
+  integration:
+    name: QEMU integration tests
+    strategy:
+      fail-fast: false
+      matrix:
+        # macos-13-large is used as macos-13 seems too flaky.
+        # macos-14 (ARM) and later cannot be used for the most part of the job
+        # due to the lack of the support for nested virt.
+        platform:
+          - macos-13-large
+          - macos-14-large
+          - macos-15-large
```

Comment:
QEMU test is quite lightweight and should be executable on ARM Mac with QEMU TCG mode

Reply:
So, you can just merge qemu.yaml to build.yaml

Reply:
I can switch to arm, but I don't want to mix integration tests with trivial build. This test is likely to break in the future and will requires workaround, lets keep it separate.

Reply:
Updated to run with arm64 machines, kept as separate workflow for easier future maintainance.